homeassistant/components/gree/__init__.py

```diff
@@ -40,23 +44,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         )
         devices.append(device)
 
-    hass.data[DOMAIN]["devices"] = devices
-    hass.data[DOMAIN]["pending"] = devices
+    coordinators = [DeviceDataUpdateCoordinator(hass, d) for d in devices]
+    await asyncio.gather(*[x.async_refresh() for x in coordinators])
+
+    hass.data[DOMAIN]["devices"] = coordinators
```

Comment:
What is the rationale for assigning the coordinators to 3 different variables in hass.data? I can't tell from reading this or each individual platform implementation. It looks like they get removed from data, but i can't tell the advantage of doing that vs just reading the list and creating an entity for each one.

Reply:
None at all, I caught this in a subsequent change and refactored already.